File: Exercism/c/space-age/src/space_age.c
#include <stdio.h>
#include "space_age.h"

static double EARTH_YEAR = 31557600.0;

double convert_planet_age(Planet planet, double seconds) {
    switch (planet) {
    case earth:
        return seconds / EARTH_YEAR;

    case mercury:
        return seconds / (EARTH_YEAR * 0.2408467);

    case venus:
        return seconds / (EARTH_YEAR * 0.61519726);

    case mars:
        return seconds / (EARTH_YEAR * 1.8808158);

    case jupiter:
        return seconds / (EARTH_YEAR * 11.862615);

    case saturn:
        return seconds / (EARTH_YEAR * 29.447498);

    case uranus:
        return seconds / (EARTH_YEAR * 84.016846);

    case neptune:
        return seconds / (EARTH_YEAR * 164.79132);
    }

    return 0;
}

File: Exercism/dart/space-age/lib/space_age.dart

class SpaceAge {
  static const EARTH_YEAR =  31557600.0;

  var years = {
    'Earth' :  EARTH_YEAR,
    'Mercury': EARTH_YEAR * 0.2408467,
    'Venus': EARTH_YEAR * 0.61519726,
    'Mars': EARTH_YEAR * 1.8808158,
    'Jupiter': EARTH_YEAR * 11.862615,
    'Saturn': EARTH_YEAR * 29.447498,
    'Uranus': EARTH_YEAR * 84.016846,
    'Neptune': EARTH_YEAR * 164.79132,
  };

  double age({String planet, int seconds}) {
    return double.parse((seconds / years[planet]).toStringAsFixed(2));
  }
}
